classes/monograph/ChapterDAO.inc.php

```diff
@@ -83,17 +83,41 @@ public function getByContextId($pressId)
         return new DAOResultFactory(
             $this->retrieve(
                 'SELECT	spc.*
-				FROM submission_chapters spc
-				INNER JOIN publications p ON (spc.publication_id = p.publication_id)
-				INNER JOIN submissions s ON (p.submission_id = s.submission_id)
-				WHERE s.context_id = ?',
+                FROM submission_chapters spc
+                INNER JOIN publications p ON (spc.publication_id = p.publication_id)
+                INNER JOIN submissions s ON (p.submission_id = s.submission_id)
+                WHERE s.context_id = ?',
                 [(int) $pressId]
             ),
             $this,
             '_fromRow'
         );
     }
 
+    /**
+     * Retrieve all chapters by source chapter id.
+     *
+     * @param $sourceChapterId int
+     * @param $orderBySequence boolean
+     *
+     * @return DAOResultFactory
+     */
+    public function getBySourceChapterId(int $sourceChapterId, bool $orderByPublicationId = true) : DAOResultFactory
+    {
+        return new DAOResultFactory(
+            $this->retrieve(
+                'SELECT	*
+                FROM submission_chapters
+                WHERE source_chapter_id = ?
+                OR chapter_id = ?'
```

Comment:
I think this should only return a result when the `chapter_id` matches when `source_chapter_id` is also `null`. Here is what I found in my example. I had two versions of a chapter with ids `1` and `2`.

- If I go to `/book/1/chapter/1` it showed me chapter 2. 
- If I go to `/book/1/chapter/2` it showed me chapter 2. However, this should only be available at `/book/1/version/2/chapter/2`.

It might not seem like it matters too much at this point, because in both cases the right content is shown. But when I add a new version of the chapter, `3`, I get the following:


- If I go to `/book/1/chapter/1` it showed me chapter 3. 
- If I go to `/book/1/chapter/2` I get a 404.
- If I go to `/book/1/chapter/3` it showed me chapter 3.

I think if we change this DAO method to only use `chapter_id` when `source_chapter_id` is also `null`, then I think our URLs will work appropriately. The `/book/1/chapter/<id>` url will always require the original chapter ID.

Does that sound good?

Reply:
Yes, I change it.